utils/api.go

```diff
@@ -0,0 +1,57 @@
+package utils
```

Comment:
This is not a generic utils, types and functionality is very specific to auth pkg. Can you  please move this into auth pkg?

Reply:
Ack.

Reply:
This is not yet done right?

Reply:
Its done. `api.go` now has all generic functions.  We can't move anything else to the auth package